test/v2/xyk/xyk-pallet.single.scenarios.test.ts

```diff
@@ -0,0 +1,771 @@
+  let user2: User;
+
+  beforeAll(async () => {
+    await setupApi();
+    [user1, user2] = setupUsers();
+
+    const { xykPalletAddress } = getEnvironmentRequiredVars();
+    xykPalletUser = new User(keyring);
+    xykPalletUser.addFromAddress(keyring, xykPalletAddress);
+
+    assetId1 = await getNextAssetId();
+    assetId2 = assetId1.add(BN_ONE);
+    liquidityAssetId = assetId2.add(BN_ONE);
+    user1.addAsset(assetId1, Assets.DEFAULT_AMOUNT);
+    user1.addAsset(assetId2, Assets.DEFAULT_AMOUNT);
+    user1.addAsset(liquidityAssetId);
+    user2.addAsset(assetId1);
+    user2.addAsset(assetId2);
+    user2.addAsset(liquidityAssetId);
+    xykPalletUser.addAsset(assetId1);
+    xykPalletUser.addAsset(assetId2);
+
+    await Sudo.batchAsSudoFinalized(
+      Assets.mintNative(user1),
+      Assets.mintNative(user2),
+      Assets.issueToken(user1),
+      Assets.issueToken(user1)
+    );
+
+    // remove native token, convenience for comparisons
+    user1.assets.pop();
+    user2.assets.pop();
+  });
+
+  beforeEach(async () => {
+    await user1.refreshAmounts(AssetWallet.BEFORE);
+    await user2.refreshAmounts(AssetWallet.BEFORE);
+    await xykPalletUser.refreshAmounts(AssetWallet.BEFORE);
+    poolBalanceBefore = await getBalanceOfPool(assetId1, assetId2);
+    totalLiquidityAssetsBefore = await getAssetSupply(liquidityAssetId);
+  });
+
+  it("xyk-pallet: create pool", async () => {
+    const assetAmount1 = new BN(50000);
+    const assetAmount2 = new BN(50000);
+
+    await signSendFinalized(
+      Xyk.createPool(assetId1, assetAmount1, assetId2, assetAmount2),
+      user1
+    );
+
+    await user1.refreshAmounts(AssetWallet.AFTER);
+    await user2.refreshAmounts(AssetWallet.AFTER);
+    await xykPalletUser.refreshAmounts(AssetWallet.AFTER);
+    const liquidityAssetsMinted = calculateLiqAssetAmount(
+      assetAmount1,
+      assetAmount2
+    );
+
+    expect([
+      user1.getAsset(assetId1)!.amountBefore.free.sub(assetAmount1),
+      user1.getAsset(assetId2)!.amountBefore.free.sub(assetAmount2),
+      user1
+        .getAsset(liquidityAssetId)!
+        .amountBefore.free.add(liquidityAssetsMinted),
+    ]).collectionBnEqual(assetsAfterFree(user1));
+
+    expect(assetsBeforeFree(user2)).collectionBnEqual(assetsAfterFree(user2));
+
+    expect([
+      xykPalletUser.getAsset(assetId1)!.amountBefore.free.add(assetAmount1),
+      xykPalletUser.getAsset(assetId2)!.amountBefore.free.add(assetAmount2),
+    ]).collectionBnEqual(assetsAfterFree(xykPalletUser));
+
+    const poolBalance = await getBalanceOfPool(assetId1, assetId2);
+    expect([
+      poolBalanceBefore[0].add(assetAmount1),
+      poolBalanceBefore[1].add(assetAmount2),
+    ]).collectionBnEqual(poolBalance);
+
+    const totalLiquidityAssets = await getAssetSupply(liquidityAssetId);
+    expect(totalLiquidityAssetsBefore.add(liquidityAssetsMinted)).bnEqual(
+      totalLiquidityAssets
+    );
+  });
+
+  it("xyk-pallet: mint liquidity", async () => {
```

Comment:
why are you splittign the test? 
Would not the second test fail if first fail?  pool wont exist. 
If that's the case, plz leave the test as it was before.

Reply:
it is much easier to maintain if it is split into logical chunks instead all in one long test
and the report looks a bit better to see what is going on

but yes, the next tests will also fail, the tests interfere with each other, so technically it should be one big test

it was a compromise of readability/maintainability vs technicality
as for the run time, i can add asserts at the start of the tests to fail early with proper message eg. pool does not exist

wdyt ?  if you insist, i can put it back in one test

Reply:
Plz,  if you don't mind :pray: 
Its a really bad practice to have tests that depend on the outcome of the others ( even more than long tests )